tests/mock_flags.py

```diff
@@ -24,43 +22,67 @@
 FLAGS = flags.FLAGS
 
 
+class MockFlag(object):
+  """Mock version of a Flag object."""
+
+  def __init__(self):
+    self.present = False
+    self.validators = ()
+    self.value = None
+
+  def Parse(self, argument):
+    self.present = True
+    self.value = argument
+
+
+class MockFlagDict(dict):
+  """Mock version of the internal dictionary of a FlagValues object.
+
+  Typically, the internal dictionary of a FlagValues object maps each flag name
+  string to a Flag object. This mock maps each flag name to a MockFlag object
+  that is created upon first access.
+
+  Because FlagValues and MockFlags override __getattr__ to search their flag
+  dictionaries, care is taken not to create a MockFlag when searching for
+  special attributes like the __copy__ method.
+  """
+
+  def __getitem__(self, key):
+    if not dict.__contains__(self, key) and not key.startswith('__'):
```

Comment:
For some reason, I was under the impression that super didn't work with dict subclasses. I tried it, and it looks like it does work, so I've changed them all to super.